sstables/sstable_set.cc

```diff
@@ -416,6 +418,10 @@ class time_series_sstable_set : public sstable_set_impl {
     schema_ptr _schema;
     lw_shared_ptr<container_t> _sstables;
 
+    min_position_reader_queue make_min_position_reader_queue(
+        std::function<flat_mutation_reader(sstable&)> create_reader,
```

Comment:
I will note again that I don't at all like reader being mixed in here.

Reply:
I agree, but if every time we try to make an optimized sstable ordering we end up having to add a new type of incremental_selector (the original used ring positions, a new one will use clustering keys, you mentioned the timestamp-ordering one), then maybe it's the wrong interface.

Reply:
aha